### PR TITLE
BUGFIX:MSF-22958 - Lookup user in domain if not found in project

### PR DESCRIPTION
**Pull request by sangtm**

### CODE REVIEW

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/1060/ : SUCCESS in 6s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-build-pipeline/944/ : SUCCESS in 8m 10s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/1061/ : SUCCESS in 6s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-build-pipeline/945/ : SUCCESS in 8m 08s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/1062/ : SUCCESS in 6s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-build-pipeline/946/ : SUCCESS in 8m 03s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/1064/ : SUCCESS in 6s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-build-pipeline/948/ : SUCCESS in 8m 26s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/1067/ : SUCCESS in 6s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-build-pipeline/951/ : SUCCESS in 8m 24s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/1068/ : SUCCESS in 7s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-build-pipeline/952/ : SUCCESS in 8m 01s

**Comment by sangtm** [comment]
ok to test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/1077/ : SUCCESS in 6s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-build-pipeline/961/ : SUCCESS in 9m 07s

File: lib/gooddata/models/user_filters/user_filter_builder.rb [hunk 1]
```diff
@@ -294,8 +338,7 @@ def self.maqlify_filters(filters, project_users, options = {})
         expression, errors = create_expression(f, labels_cache, lookups_cache, attrs_cache, options)
         safe_login = login.downcase
         profiles_uri = if options[:type] == :muf
-                         project_user = project_users.find { |u| u.login == safe_login }
-                         project_user.nil? ? ('/gdc/account/profile/' + safe_login) : project_user.profile_url
+                         user_profile_mapping[safe_login]
```
Comment (danh-ung) [inline lib/gooddata/models/user_filters/user_filter_builder.rb:341]: what happened if user not in the domain.users, we should log as error and skip that user.

File: lib/gooddata/models/user_filters/user_filter_builder.rb [hunk 2]
```diff
@@ -584,12 +628,12 @@ def self.normalize_filters(filters)
     # Removes MUFs from to_delete unless in user is in users_brick_input
     # if this does not happen, users that are about to be deleted by users_brick
     # would have all their filters removed now, which is not desirable
-    def self.sanitize_filters_to_delete(to_delete, users_brick_input, project_users)
+    def self.sanitize_filters_to_delete(to_delete, users_brick_input, user_profile_mapping)
       return [] unless users_brick_input && users_brick_input.any?
       user_profiles = users_brick_input.map do |user|
-        result = project_users.find { |u| u.login == user.with_indifferent_access['login'] }
+        result = user_profile_mapping[user.with_indifferent_access['login']]
```
Comment (danh-ung) [inline lib/gooddata/models/user_filters/user_filter_builder.rb:634]: user_profile_mapping don't contains user of deleted filters, only input_filters.

File: lib/gooddata/models/user_filters/user_filter_builder.rb (hunk 1 above)
Comment (sangtm) [reply lib/gooddata/models/user_filters/user_filter_builder.rb:341]: Changed to use the login name if not found, thanks

File: lib/gooddata/models/user_filters/user_filter_builder.rb (hunk 2 above)
Comment (sangtm) [reply lib/gooddata/models/user_filters/user_filter_builder.rb:634]: The delete MUF will be calculated from users_brick_input, to_delete and input_filters.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/1083/ : SUCCESS in 8s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-build-pipeline/967/ : SUCCESS in 8m 36s

**Comment by sangtm** [comment]
ok to test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/1087/ : SUCCESS in 9s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-build-pipeline/971/ : SUCCESS in 8m 05s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/1089/ : SUCCESS in 8s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-build-pipeline/973/ : SUCCESS in 10m 14s

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ruby-gate-PR1848

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-sonarqube-scan-zuul-gate/74/ : SUCCESS in 2m 01s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-merge-compliance-check/616/ : SUCCESS in 24s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-gitops-deploy-pipeline/40/ : SUCCESS in 1m 43s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-fossa-scanning-zuul-el7/159/ : SUCCESS in 1m 45s (non-voting)

**Pull request merged**